File: src/de/saschahlusiak/freebloks/network/NET_UNDO_STONE.java
package de.saschahlusiak.freebloks.network;

public class NET_UNDO_STONE extends NET_SET_STONE {

	public NET_UNDO_STONE() {
		super(Network.MSG_UNDO_STONE);
	}

	public NET_UNDO_STONE(NET_HEADER from) {
		super(from);
	}
}

File: src/main/java/de/saschahlusiak/freebloks/network/message/NetUndoStone.kt
package de.saschahlusiak.freebloks.network.message

import de.saschahlusiak.freebloks.network.*
import java.nio.ByteBuffer

class NetUndoStone: Message(MessageType.UndoStone) {
    companion object {
        fun from(data: ByteBuffer): NetUndoStone {
            // BUG: MSG_UNDO_STONE is actually a NET_SET_STONE with random payload; don't verify
            repeat(6) { data.get() }

            return NetUndoStone()
        }
    }
}